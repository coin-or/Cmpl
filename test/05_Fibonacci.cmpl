
// two methods for computing fibonacci numbers

echo ("computing fibonacci by recursive function");

fib := &+{ $arg <= 2: 1 |: fib($arg-1) + fib($arg-2) };
{ i in 1..20: echo(i + ": " + fib(i)); }



echo ("");
echo ("computing fibonacci by a class storing the already computed numbers");

fibcl := &{:
	private par:
		$this._fib[1..2] := 1;
		$this._maxind := 2;

		$this.compute := &{ $arg > $this._maxind:
				{ i in ($this._maxind+1) .. $arg: $this._fib[i] := $this._fib[i-1] + $this._fib[i-2]; }
			};

	public par:
		$this.get := &{:
				{ $arg > $this._maxind: $this.compute($arg); }
				return $this._fib[$arg];
			};
};

fibobj := class.construct(fibcl);

{ i in 1..20: echo(i + ": " + fibobj.get(i)); }

