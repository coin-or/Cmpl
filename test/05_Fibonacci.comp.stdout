computing fibonacci by recursive function
1: 1
2: 1
3: 2
4: 3
5: 5
6: 8
7: 13
8: 21
9: 34
10: 55
11: 89
12: 144
13: 233
14: 377
15: 610
16: 987
17: 1597
18: 2584
19: 4181
20: 6765

computing fibonacci by a class storing the already computed numbers
1: 1
2: 1
3: 2
4: 3
5: 5
6: 8
7: 13
8: 21
9: 34
10: 55
11: 89
12: 144
13: 233
14: 377
15: 610
16: 987
17: 1597
18: 2584
19: 4181
20: 6765
